server/src/main/java/org/opensearch/index/translog/TranslogStats.java

```diff
@@ -159,6 +178,69 @@ public RemoteTranslogStats getRemoteTranslogStats() {
         return remoteTranslogStats;
     }
 
+    /**
+     * Builder for the {@link TranslogStats} class.
+     * Provides a fluent API for constructing a TranslogStats object.
+     */
+    public static class Builder {
+        private int numberOfOperations = 0;
+        private long translogSizeInBytes = 0;
+        private int uncommittedOperations = 0;
+        private long uncommittedSizeInBytes = 0;
+        private long earliestLastModifiedAge = 0;
+        private final RemoteTranslogStats remoteTranslogStats = new RemoteTranslogStats();
+
+        public Builder() {}
+
+        public Builder numberOfOperations(int operations) {
+            this.numberOfOperations = operations;
+            return this;
+        }
+
+        public Builder translogSizeInBytes(long size) {
+            this.translogSizeInBytes = size;
+            return this;
+        }
+
+        public Builder uncommittedOperations(int operations) {
+            this.uncommittedOperations = operations;
+            return this;
+        }
+
+        public Builder uncommittedSizeInBytes(long bytes) {
+            this.uncommittedSizeInBytes = bytes;
+            return this;
+        }
+
+        public Builder earliestLastModifiedAge(long age) {
+            this.earliestLastModifiedAge = age;
+            return this;
+        }
+
+        /**
+         * Creates a {@link TranslogStats} object from the builder's current state.
+         * @return A new TranslogStats instance.
+         */
+        public TranslogStats build() {
+            if (numberOfOperations < 0) {
+                throw new IllegalArgumentException("numberOfOperations must be >= 0");
+            }
+            if (translogSizeInBytes < 0) {
+                throw new IllegalArgumentException("translogSizeInBytes must be >= 0");
+            }
+            if (uncommittedOperations < 0) {
+                throw new IllegalArgumentException("uncommittedOperations must be >= 0");
+            }
+            if (uncommittedSizeInBytes < 0) {
+                throw new IllegalArgumentException("uncommittedSizeInBytes must be >= 0");
+            }
+            if (earliestLastModifiedAge < 0) {
+                throw new IllegalArgumentException("earliestLastModifiedAge must be >= 0");
+            }
```

Comment:
nit: you can actually pull these checks in the setters instead to throw exception early. Basically don't let anyone set the values in the builder in the first place before waiting to build.